assistant/v2.ts

```diff
@@ -76,11 +76,19 @@ class AssistantV2 extends BaseService {
    * @param {Function} [callback] - The callback that handles the response.
    * @returns {NodeJS.ReadableStream|void}
    */
-  public createSession(params: AssistantV2.CreateSessionParams, callback?: AssistantV2.Callback<AssistantV2.SessionResponse>): NodeJS.ReadableStream | void {
+  public createSession(params: AssistantV2.CreateSessionParams, callback?: AssistantV2.Callback<AssistantV2.SessionResponse>): NodeJS.ReadableStream | Promise<any> | void {
     const _params = extend({}, params);
-    const _callback = (callback) ? callback : () => { /* noop */ };
+    const _callback = callback;
     const requiredParams = ['assistant_id'];
 
+    if (!_callback) {
+      return new Promise((resolve, reject) => {
+        this.createSession(params, (err, bod, res) => {
+          err ? reject(err) : _params.return_response ? resolve(res) : resolve(bod);
```

Comment:
I would prefer to always return `body` and `response` as part of the response.
We are currently moving all the other SDKs to always return the HTTP response and the parsed body. I can't remember the names we are using but @lpatino10 or @ehdsouza can probably help you.

Reply:
The issue is that a Promise can only resolve _one_ value. Should I be wrapping the whole respond and the parsed body up into one object and return that? Or just the response and let the user parse the body?

Reply:
I don't think I understand. You're saying to return a single object that contains the results and the headers?

e.g.
```json
{
  "result": {
    "<body>"
  },
  "headers": {
    "<response headers>"
  }
}
```

We currently also return the request information which I think is beneficial for debugging, etc.

Why not just return the entire response if users have to go through an additional step to get to the body anyway?

Reply:
The entire response depends on the library that we currently use (`requests`) my proposal is to define the same interface we use with other sdks.

```json
{
  "result": <body-as-json-or-buffer>,
  "headers": {
    "header1": "value1"
  }
}
```

Reply:
That's a good point. I'll wait to finish this PR until I finish converting the SDK to use `axios` instead of `request`.

Would you be okay with also returning the request object if possible? I find it can be helpful for debugging to see what the request looked like when it was sent:
```json
{
  "result": "<body-as-json-or-buffer>",
  "headers": {
    "header1": "value1"
  },
  "request": " <request object>"
}
```

Reply:
The default response structure for `axios` looks like this:
```
{
  // `data` is the response that was provided by the server
  data: {},

  // `status` is the HTTP status code from the server response
  status: 200,

  // `statusText` is the HTTP status message from the server response
  statusText: 'OK',

  // `headers` the headers that the server responded with
  // All header names are lower cased
  headers: {},

  // `config` is the config that was provided to `axios` for the request
  config: {},

  // `request` is the request that generated this response
  // It is the last ClientRequest instance in node.js (in redirects)
  // and an XMLHttpRequest instance the browser
  request: {}
}
```

We might want to keep all of that information and rename `data` to `result` for consistency

Reply:
as long as we can change the library without breaking the interface